src/libs/ascent/utils/ascent_string_utils.cpp

```diff
@@ -39,39 +50,133 @@ void split_string(const std::string &s,
 
 } // namespace detail
 
-std::string expand_family_name(const std::string name, int counter)
+template<typename T>
+std::string expand_format_value(const std::string path_string,
+                                const T value)
 {
-  if(counter == 0)
+  std::string result_string = path_string;
+
+  // Maintaing legacy family string formatting
+  bool has_format = result_string.find("%") != std::string::npos;
+  if(has_format)
+  {
+    // allow for long file paths
+    char buffer[2048];
+    snprintf(buffer, 2048, result_string.c_str(), value);
+    result_string = std::string(buffer);
+  }
+
+  return result_string;
+}
+
+template<typename T>
+std::string expand_generic_variable(const std::string& path_string, 
+                                    const std::regex& pattern, 
+                                    const T value)
+{
+    std::smatch match;
+    std::string result_string = path_string;
+
+    std::regex valid_int_format(R"(^\d*d$)");
```

Comment:
these look good to me, as a gift to our future selves, can you add a comment  about the types of patterns these will catch?